topics/transcriptomics/tutorials/ref-based/tutorial.md

```diff
@@ -1561,43 +1813,65 @@ We have extracted genes that are differentially expressed in treated (PS gene-de
+> 1. {% tool [Compute](toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/1.6) %} with the following parameters:
 >    - *"Add expression"*: `bool(c7<0.05)`
->    - {% icon param-file %} *"as a new column to"*: the `DESeq2 result file`
+>    - {% icon param-file %} *"as a new column to"*: the `DESeq2 result file` (output of **DESeq2** {% icon tool %})
 >
 > 2. {% tool [Cut](Cut1) %} columns from a table with the following parameters:
 >    - *"Cut columns"*: `c1,c8`
 >    - *"Delimited by"*: `Tab`
->    - {% icon param-file %} *"From"*: the output of the **Compute**
+>    - {% icon param-file %} *"From"*: the output of the **Compute** {% icon tool %}
 >
 > 3. {% tool [Change Case](ChangeCase) %} with
->    - {% icon param-file %} *"From"*: the output of the previous **Cut**
+>    - {% icon param-file %} *"From"*: the output of the previous **Cut** {% icon tool %}
 >    - *"Change case of columns"*: `c1`
 >    - *"Delimited by"*: `Tab`
 >    - *"To"*: `Upper case`
 >
 > 4. Rename the output to `Gene IDs and differential expression`.
+{: .hands_on}
+
+We just generated the first input for **goseq**. As second input for **goseq** we need the gene lengths. We can use here the gene lengths generated by **featureCounts**  or **Gene length and GC content** and format the gene IDs.
+
+> ### {% icon hands_on %} Hands-on: Get the gene length from previous history
+>
+> <div class="featureCounts" markdown="1">
+> 1. Copy the feature length collection previously generated by **featureCounts** {% icon tool %} into this history
 >
->    We just generated the first input for **goseq**. As second input for **goseq** we need the gene lengths. We can use here the gene lengths generated by **featureCounts** and format the gene IDs.
+>    {% snippet faqs/galaxy/histories_copy_dataset.md %}
 >
-> 5. Drag and drop one of the feature length datasets generated by **featureCounts** {% icon tool %} into this history using the {% icon galaxy-columns %} **View all histories**.
+> 2. {% tool [Extract Dataset](__EXTRACT_DATASET__) %} with:
+>   - {% icon param-collection %} *"Input List"*: `featureCounts on collection N: Feature lengths`
+>   - *"How should a dataset be selected?"*: `The first dataset`
+> </div>
 >
-> 6. {% tool [Change Case](ChangeCase) %} with
->    - {% icon param-file %} *"From"*: the `feature lengths` (output of **featureCounts**)
+> <div class="STAR" markdown="1">
+> 1. Copy the output of **Gene length and GC content** {% icon tool %} (`Gene length`) into this history
+>
+>   {% snippet faqs/galaxy/histories_copy_dataset.md %}
+> </div>
+>
+{: .hands_on}
+
+> ### {% icon hands_on %} Hands-on: Adapt the gene length to goseq
+>
+> 1. {% tool [Change Case](ChangeCase) %} with the following parameters:
+>
+>    - {% icon param-file %} *"From"*: <span class="featureCounts" markdown="1">`GSM461177_untreat_paired` (output of **Extract Dataset** {% icon tool %})</span><span class="STAR" markdown="1">`Gene length`</span>
```

Comment:
Probably there is a good reason for that, but I cannot understand why you are using HTML code here.

Reply:
it's for the choose your own tutorial feature.